server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java

```diff
@@ -0,0 +1,671 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.coordination;
+
+import com.carrotsearch.randomizedtesting.RandomizedContext;
+
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.gateway.ClusterStateUpdaters;
+import org.elasticsearch.test.junit.annotations.TestLogging;
+import org.elasticsearch.threadpool.Scheduler;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.After;
+import org.junit.Before;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.function.BooleanSupplier;
+import java.util.function.Function;
+import java.util.function.LongSupplier;
+
+import static org.elasticsearch.cluster.coordination.CoordinationStateTests.clusterState;
+
+@TestLogging(reason = "these tests do a lot of log-worthy things but we usually don't care", value = "org.elasticsearch:FATAL")
+public class AtomicRegisterCoordinatorTests extends CoordinatorTests {
+    private AtomicRegister atomicRegister;
+    private SharedStore sharedStore;
```

Comment:
I think these things should be scoped within a single cluster rather than here at the top level. I see that this is going to be a more substantial change tho, because we call `Cluster::new` in lots of places today that would need adjusting. And we're suppressing all the tests that involve multiple clusters at the moment too. Maybe just a TODO comment and we can handle this in a followup.